app/src/main/java/org/mozilla/fenix/settings/CookieBannerFragment.kt

```diff
@@ -0,0 +1,131 @@
+import org.mozilla.fenix.utils.view.addToRadioGroup
+
+/**
+ * Lets the user set up the cookie banner handling preferences.
+ */
+class CookieBannerFragment : PreferenceFragmentCompat() {
+
+    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
+        setPreferencesFromResource(R.xml.cookie_banner_preferences, rootKey)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        showToolbar(getString(R.string.preferences_cookie_banner_reduction))
+        setupPreferences()
+    }
+
+    private fun getEngineSettings(): Settings {
+        return requireContext().components.core.engine.settings
+    }
+
+    private fun setupPreferences() {
+        requirePreference<CheckBoxPreference>(R.string.pref_key_cookie_banner_private_browsing).apply {
+            isChecked = context.settings().cookieBannerHandlingPrivateBrowsing
+            onPreferenceChangeListener = object : SharedPreferenceUpdater() {
+                override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                    val cookieBannerModePrivateBrowsing =
+                        requireContext().settings().getCookieBannerHandlingPrivateBrowsing()
+                    val (privateMode, normalMode) = when (newValue as Boolean) {
+                        true -> {
+                            Pair(cookieBannerModePrivateBrowsing, CookieBannerHandlingMode.DISABLED)
+                        }
+                        false -> {
+                            Pair(cookieBannerModePrivateBrowsing, cookieBannerModePrivateBrowsing)
+                        }
+                    }
+
+                    getEngineSettings().cookieBannerHandlingModePrivateBrowsing = privateMode
+                    getEngineSettings().cookieBannerHandlingMode = normalMode
+
+                    return super.onPreferenceChange(preference, newValue)
+                }
+            }
+        }
+
+        val rejectAllCookies =
+            requirePreference<RadioButtonPreference>(R.string.pref_key_cookie_banner_reject_all).apply {
+                onPreferenceChangeListener = object : SharedPreferenceUpdater() {
+                    override fun onPreferenceChange(preference: Preference, newValue: Any?): Boolean {
+                        if (newValue == true) {
+                            // we always change the private browsing setting, regardless of the checkbox state
+                            getEngineSettings().cookieBannerHandlingModePrivateBrowsing =
+                                CookieBannerHandlingMode.REJECT_ALL
+
+                            // change for normal browsing only if the checkbox is off
+                            if (!context.settings().cookieBannerHandlingPrivateBrowsing) {
+                                getEngineSettings().cookieBannerHandlingMode =
+                                    CookieBannerHandlingMode.REJECT_ALL
+                            }
+                        }
```

Comment:
We could take this redundant code out into a function:

```
fun setCookiePrefHelper (mode) {
    // we always change the private browsing setting, regardless of the checkbox state
    getEngineSettings().cookieBannerHandlingModePrivateBrowsing = mode

    // change for normal browsing only if the checkbox is off
    if (!context.settings().cookieBannerHandlingPrivateBrowsing) {
        getEngineSettings().cookieBannerHandlingMode = mode
    }
}
```
Would be good to do anyway, in case ui team changes this logic

Reply:
👍🏽

Reply:
Please feel free to push any improvements!